Source/MQTTnet/Packets/MqttPublishPacket.cs

```diff
@@ -20,7 +21,7 @@ public sealed class MqttPublishPacket : MqttPacketWithIdentifier
 
     public MqttPayloadFormatIndicator PayloadFormatIndicator { get; set; } = MqttPayloadFormatIndicator.Unspecified;
 
-    public ArraySegment<byte> PayloadSegment { get; set; }
+    public ReadOnlySequence<byte> PayloadSequence { get; set; }
```

Comment:
Consider renaming this to "Payload" because the Type suffix is no longer needed.